### ISSUE TITLE
SumLazyTensor supports adding torch tensors [Feature Request]

### ISSUE THREAD

**Issue by wjmaddox** [cornellius-gp/gpytorch#879]
# 🚀 Feature Request

SumLazy Tensor should be able to handle adding tensors like NonLazyTensors do.

## Motivation

**Is your feature request related to a problem? Please describe.**
I saw this while attempting to have enforced jitter in the covariance in a variational strategy.

## Pitch

**Describe the solution you'd like**
SumLazyTensor allows the following code to work - should be a matter of adding another if statement to the __add__ method.

```
test1 = gpytorch.lazify(torch.randn(30, 30))

test2 = torch.randn(30,30)
res = test1 + test2
res + test2 # currently does not work but should work.
```

**Describe alternatives you've considered**

We could enforce both tensors here to be lazy tensors in the variational strategy instead. See [here](https://github.com/cornellius-gp/gpytorch/blob/e50b9878a090b41eaf38f371cab2b938d5b2ebbc/gpytorch/variational/variational_strategy.py#L216)

[NonLazyTensor](https://github.com/cornellius-gp/gpytorch/blob/e50b9878a090b41eaf38f371cab2b938d5b2ebbc/gpytorch/lazy/non_lazy_tensor.py#L71)  does allow this type of addition.
**Are you willing to open a pull request?**
Yes.

## Additional context

See proposed code above.

**Comment by jacobrgardner** [comment]
Yeah, this is reasonable. In general, I think we should just have `__add__` and `__radd__` in LazyTensor itself support `LT + T` and `T + LT` ops.

The one "gotcha" is we'll need to special case things so that broadcasting still works -- e.g. `torch.randn(30, 1) + SomeArbitraryLazyTensor(30, 30)` is not as simple as `return SumLazyTensor(NonLazyTensor(torch.randn(30, 1), SomeArbitraryLazyTensor(30, 30))`. The first LT needs to be wrapped in some kind of `BroadcastingLazyTensor` whose _matmul does the equivalent of `torch.randn(30, 1).repeat(1, 30).matmul(...)`.

**Comment by wjmaddox** [comment]
Good catch on the broadcasting semantics. It looks like we can use `torch.distributions.utils.broadcast_all` on the shapes to either build the BroadcastingLazyTensor properly or to assign it within the SumLazyTensor itself.
